commands/project/profile/add.ts

```diff
@@ -0,0 +1,268 @@
+      if (validationResult !== true) {
+        uiLogger.error(validationResult);
+        uiLogger.log('');
+        profileName = undefined;
+      }
+    }
+  }
+
+  if (!profileName) {
+    const promptResponse = await promptUser<{ name: string }>({
+      type: 'input',
+      name: 'name',
+      message: commands.project.profile.add.prompts.namePrompt,
+      validate: isValidProfileName,
+    });
+
+    profileName = promptResponse.name;
+  }
+
+  if (targetAccount) {
+    const accountId = getAccountId(targetAccount);
+    if (accountId) {
+      targetAccount = accountId;
+    } else {
+      uiLogger.error(commands.project.profile.add.errors.invalidTargetAccount);
+      uiLogger.log('');
+      targetAccount = undefined;
+    }
+  }
+
+  if (!targetAccount) {
+    const configuredAccounts = getConfigAccounts();
+
+    if (!configuredAccounts || !configuredAccounts.length) {
+      uiLogger.error(commands.project.profile.add.errors.noAccountsConfigured);
+      process.exit(EXIT_CODES.ERROR);
+    }
+
+    const promptResponse = await listPrompt(
+      commands.project.profile.add.prompts.targetAccountPrompt,
+      {
+        choices: configuredAccounts.map(account => {
+          const accountId = getAccountIdentifier(account);
+          return {
+            name: uiAccountDescription(accountId),
+            value: accountId,
+          };
+        }),
+      }
+    );
+
+    if (promptResponse) {
+      targetAccount = promptResponse;
+    }
+  }
+
+  const profileFileContent: HsProfileFile = {
+    accountId: Number(targetAccount),
+    variables: {},
+  };
+
+  const existingProfiles = await getAllHsProfiles(projectSourceDir);
+  let profileToCopyVariablesFrom: string | undefined;
+
+  if (existingProfiles.length == 1) {
+    uiLogger.log('');
+    uiLogger.log(
+      commands.project.profile.add.logs.copyExistingProfile(
+        getHsProfileFilename(existingProfiles[0])
+      )
+    );
+    const shouldCopyVariables = await confirmPrompt('Copy profile variables?', {
+      defaultAnswer: true,
+    });
+
+    if (shouldCopyVariables) {
+      profileToCopyVariablesFrom = existingProfiles[0];
+    }
+  } else if (existingProfiles.length > 1) {
+    uiLogger.log('');
+    uiLogger.log(commands.project.profile.add.logs.copyExistingProfiles);
+    const emptyChoice = {
+      name: commands.project.profile.add.prompts.copyExistingProfilePromptEmpty,
+      value: undefined,
+    };
+
+    const promptResponse = await listPrompt(
+      commands.project.profile.add.prompts.copyExistingProfilePrompt,
+      {
+        choices: [
+          ...existingProfiles.map(profile => ({
+            name: getHsProfileFilename(profile),
+            value: profile,
+          })),
+          emptyChoice,
+        ],
+      }
+    );
+
+    if (promptResponse) {
+      profileToCopyVariablesFrom = promptResponse;
+    }
+  }
+
+  if (profileToCopyVariablesFrom) {
+    try {
+      const profileToCopyFileContent = loadHsProfileFile(
```

Comment:
Not using `loadProfile` here because we don't want to log anything on errors. We'll make the variables section in the new config empty if this fails.